File: Leetcode/cpp/easy/1732.cpp
/*
 * @lc app=leetcode id=1732 lang=cpp
 *
 * [1732] Find the Highest Altitude
 */


// @lcpr-template-start
#include <bits/stdc++.h>
using namespace std;
// @lcpr-template-end
// @lc code=start
class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        int ans = 0, s = 0;
        for (auto& x : gain) {
            s += x;
            ans = max(ans, s);
        }
        return ans;
    }
};
// @lc code=end

File: Leetcode/rust/easy/1732.rs
/*
 * @lc app=leetcode id=1732 lang=rust
 *
 * [1732] Find the Highest Altitude
 */


// @lcpr-template-start
struct Solution;
// @lcpr-template-end
// @lc code=start
impl Solution {
    pub fn largest_altitude(gain: Vec<i32>) -> i32 {
        let (mut ans, mut s): (i32, i32) = (0, 0);
        for x in gain {
            s += x;
            ans = ans.max(s);
        }
        ans
    }
}
// @lc code=end

